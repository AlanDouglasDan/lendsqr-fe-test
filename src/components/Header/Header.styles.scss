@use "variables";

.header {
  height: 100px;
  box-shadow: 3px 0px 20px rgba(0, 0, 0, 0.04);
  background-color: variables.$white;
  padding: 0 30px;
}

.unionImage {
  width: 20px;
  margin-right: 8px;
  object-fit: contain;
}

.lendsqrImage {
  width: 115px;
  object-fit: contain;
}

.usersBody .input-group {
  width: 400px;
}

.docsText {
  font-size: 16px;
  text-decoration: underline;
  color: variables.$dark-blue;
  margin-right: 50px;
}

.bellIcon {
  color: variables.$dark-blue;
  width: 20px;
  height: 22px;
  margin-right: 33px;
}

.profileImage {
  width: 48px;
  height: 48px;
  border-radius: 50%;
  margin-right: 10px;
}

.profileText {
  font-size: 16px;
  color: variables.$dark-blue;
  margin-right: 10px;
}

.angleIcon {
  color: variables.$dark-blue;
  margin-right: 33px;
}

.hamburger {
  display: none;
}

.bar {
  display: block;
  width: 25px;
  height: 3px;
  margin: 5px auto;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  background-color: variables.$dark-blue;
}

.d-none {
  display: none !important;
}