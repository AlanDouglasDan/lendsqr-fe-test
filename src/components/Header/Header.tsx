import { Link } from "react-router-dom";

import images from "../../assets/images";
import "./Header.styles.scss";

type Props = {
  toggleMobileMenu: () => void;
  mobileMenuOpen: Boolean;
};

const Header: React.FC<Props> = ({ toggleMobileMenu, mobileMenuOpen }) => {
  return (
    <div className="header row m-0">
      <div className="col-md-6 d-flex align-items-center px-0 d-med-none">
        <Link to="/dashboard/users" className="no-underline text-dark me-5">
          <div className="d-flex align-items-start">
            <img src={images.union} alt="Union" className="unionImage" />
            <img src={images.lendsqr} alt="Lendsqr" className="lendsqrImage" />
          </div>
        </Link>

        <div className="input-group my-4 border-0 ms-3">
          <input
            type="search"
            className="form-control border-right-0"
            placeholder="Search for anything"
            aria-label="search"
            id="search"
            autoComplete="off"
          />
          <span className="input-group-text bg-theme text-white border-0">
            <i className="fa fa-search" />
          </span>
        </div>
      </div>

      <div className="col-md-6 d-flex align-items-center justify-content-end px-0 d-med-none">
        <span className="docsText cursor-pointer">Docs</span>

        <i className="fa fa-bell-o bellIcon cursor-pointer" />

        <img
          src={images.profile}
          alt="Profile"
          className="profileImage cursor-pointer"
        />

        <span className="profileText">Adedeji</span>

        <i className="fa fa-caret-down angleIcon cursor-pointer" />
      </div>

      <div className="d-noone d-med-flex align-items-center justify-content-between p-0">
        <div className="d-flex align-items-center">
          <div
            onClick={toggleMobileMenu}
            className={`hamburger d-flex align-items-center justify-content-center me-4 mt-1 ${
              mobileMenuOpen === true && "open"
            }
          `}
          >
            <span className="hamburger-top"></span>
            <span className="hamburger-middle"></span>
            <span className="hamburger-bottom"></span>
          </div>

          <Link to="/dashboard/users" className="no-underline text-dark">
            <div className="d-flex align-items-start">
              <img src={images.union} alt="Union" className="unionImage" />
              <img
                src={images.lendsqr}
                alt="Lendsqr"
                className="lendsqrImage"
              />
            </div>
          </Link>
        </div>

        <img
          src={images.profile}
          alt="Profile"
          className="profileImage cursor-pointer"
        />
      </div>
    </div>
  );
};

export default Header;
