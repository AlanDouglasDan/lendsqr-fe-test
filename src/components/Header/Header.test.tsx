import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import Header from "./Header";

describe("Header Component Test", () => {
  it("Header Component Mobile View Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header mobileMenuOpen={true} toggleMobileMenu={() => {}} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Header Component Desktop View Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header mobileMenuOpen={false} toggleMobileMenu={() => {}} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
