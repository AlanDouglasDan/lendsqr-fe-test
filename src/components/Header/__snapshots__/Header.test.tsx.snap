// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test Header Component Desktop View Snapshot 1`] = `
<DocumentFragment>
  <div
    class="header row m-0"
  >
    <div
      class="col-md-6 d-flex align-items-center justify-content-between px-0 d-med-none"
    >
      <a
        class="no-underline text-dark"
        href="/dashboard/users"
      >
        <div
          class="d-flex align-items-start"
        >
          <img
            alt="Union"
            class="unionImage"
            src="Union.png"
          />
          <img
            alt="Lendsqr"
            class="lendsqrImage"
            src="lendsqr.png"
          />
        </div>
      </a>
      <div
        class="input-group my-4 border-0"
      >
        <input
          aria-label="search"
          class="form-control border-right-0"
          placeholder="Search for anything"
          type="text"
        />
        <span
          class="input-group-text bg-theme text-white border-0"
        >
          <i
            class="fa fa-search"
          />
        </span>
      </div>
    </div>
    <div
      class="col-md-6 d-flex align-items-center justify-content-end px-0 d-med-none"
    >
      <span
        class="docsText cursor-pointer"
      >
        Docs
      </span>
      <i
        class="fa fa-bell-o bellIcon cursor-pointer"
      />
      <img
        alt="Profile"
        class="profileImage cursor-pointer"
        src="profile.png"
      />
      <span
        class="profileText"
      >
        Adedeji
      </span>
      <i
        class="fa fa-caret-down angleIcon cursor-pointer"
      />
    </div>
    <div
      class="d-noone d-med-flex align-items-center justify-content-between p-0"
    >
      <div
        class="d-flex align-items-center"
      >
        <div
          class="hamburger d-flex align-items-center justify-content-center me-4 mt-1 false
          "
        >
          <span
            class="hamburger-top"
          />
          <span
            class="hamburger-middle"
          />
          <span
            class="hamburger-bottom"
          />
        </div>
        <a
          class="no-underline text-dark"
          href="/dashboard/users"
        >
          <div
            class="d-flex align-items-center"
          >
            <img
              alt="Union"
              class="unionImage"
              src="Union.png"
            />
            <img
              alt="Lendsqr"
              class="lendsqrImage"
              src="lendsqr.png"
            />
          </div>
        </a>
      </div>
      <img
        alt="Profile"
        class="profileImage cursor-pointer"
        src="profile.png"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header Component Test Header Component Mobile View Snapshot 1`] = `
<DocumentFragment>
  <div
    class="header row m-0"
  >
    <div
      class="col-md-6 d-flex align-items-center justify-content-between px-0 d-med-none"
    >
      <a
        class="no-underline text-dark"
        href="/dashboard/users"
      >
        <div
          class="d-flex align-items-start"
        >
          <img
            alt="Union"
            class="unionImage"
            src="Union.png"
          />
          <img
            alt="Lendsqr"
            class="lendsqrImage"
            src="lendsqr.png"
          />
        </div>
      </a>
      <div
        class="input-group my-4 border-0"
      >
        <input
          aria-label="search"
          class="form-control border-right-0"
          placeholder="Search for anything"
          type="text"
        />
        <span
          class="input-group-text bg-theme text-white border-0"
        >
          <i
            class="fa fa-search"
          />
        </span>
      </div>
    </div>
    <div
      class="col-md-6 d-flex align-items-center justify-content-end px-0 d-med-none"
    >
      <span
        class="docsText cursor-pointer"
      >
        Docs
      </span>
      <i
        class="fa fa-bell-o bellIcon cursor-pointer"
      />
      <img
        alt="Profile"
        class="profileImage cursor-pointer"
        src="profile.png"
      />
      <span
        class="profileText"
      >
        Adedeji
      </span>
      <i
        class="fa fa-caret-down angleIcon cursor-pointer"
      />
    </div>
    <div
      class="d-noone d-med-flex align-items-center justify-content-between p-0"
    >
      <div
        class="d-flex align-items-center"
      >
        <div
          class="hamburger d-flex align-items-center justify-content-center me-4 mt-1 open
          "
        >
          <span
            class="hamburger-top"
          />
          <span
            class="hamburger-middle"
          />
          <span
            class="hamburger-bottom"
          />
        </div>
        <a
          class="no-underline text-dark"
          href="/dashboard/users"
        >
          <div
            class="d-flex align-items-center"
          >
            <img
              alt="Union"
              class="unionImage"
              src="Union.png"
            />
            <img
              alt="Lendsqr"
              class="lendsqrImage"
              src="lendsqr.png"
            />
          </div>
        </a>
      </div>
      <img
        alt="Profile"
        class="profileImage cursor-pointer"
        src="profile.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
